admin/pages/network.php

```diff
@@ -11,87 +11,89 @@
-
-$options = get_site_option( 'wpseo_ms' );
-
-if ( isset( $_POST['wpseo_submit'] ) ) {
-	check_admin_referer( 'wpseo-network-settings' );
+function wpseo_ms_get_site_dropdown_choices( $include_empty = false ) {
+	$choices = array();
 
-	foreach ( array( 'access', 'defaultblog' ) as $opt ) {
-		$options[ $opt ] = $_POST['wpseo_ms'][ $opt ];
+	if ( $include_empty ) {
+		$choices['-'] = __( 'None', 'wordpress-seo' );
 	}
-	unset( $opt );
-	WPSEO_Options::update_site_option( 'wpseo_ms', $options );
-	add_settings_error( 'wpseo_ms', 'settings_updated', __( 'Settings Updated.', 'wordpress-seo' ), 'updated' );
-}
 
-if ( isset( $_POST['wpseo_restore_blog'] ) ) {
-	check_admin_referer( 'wpseo-network-restore' );
-	if ( isset( $_POST['wpseo_ms']['restoreblog'] ) && is_numeric( $_POST['wpseo_ms']['restoreblog'] ) ) {
-		$restoreblog = (int) WPSEO_Utils::validate_int( $_POST['wpseo_ms']['restoreblog'] );
-		$blog        = get_blog_details( $restoreblog );
+	$available_states = array(
+		'public'   => __( 'public', 'wordpress-seo' ),
+		'archived' => __( 'archived', 'wordpress-seo' ),
+		'mature'   => __( 'mature', 'wordpress-seo' ),
+		'spam'     => __( 'spam', 'wordpress-seo' ),
+	);
 
-		if ( $blog ) {
-			WPSEO_Options::reset_ms_blog( $restoreblog );
-			/* translators: %s expands to the name of a blog within a multi-site network. */
-			add_settings_error( 'wpseo_ms', 'settings_updated', sprintf( __( '%s restored to default SEO settings.', 'wordpress-seo' ), esc_html( $blog->blogname ) ), 'updated' );
+	$sites = get_sites( array( 'deleted' => 0 ) );
+	foreach ( $sites as $site ) {
+		$choices[ $site->blog_id ] = $site->blog_id . ': ' . $site->domain;
+
+		$site_states = array();
+		foreach ( $available_states as $state_slug => $state_label ) {
+			if ( $site->$state_slug === '1' ) {
+				$site_states[] = $state_label;
+			}
 		}
-		else {
-			/* translators: %s expands to the ID of a blog within a multi-site network. */
-			add_settings_error( 'wpseo_ms', 'settings_updated', sprintf( __( 'Blog %s not found.', 'wordpress-seo' ), esc_html( $restoreblog ) ), 'error' );
+
+		if ( ! empty( $site_states ) ) {
+			$choices[ $site->blog_id ] .= ' [' . implode( ', ', $site_states ) . ']';
 		}
-		unset( $restoreblog, $blog );
 	}
-}
 
-/* Set up selectbox dropdowns for smaller networks (usability) */
-$use_dropdown = true;
-if ( get_blog_count() > 100 ) {
-	$use_dropdown = false;
+	return $choices;
 }
-else {
-
-	$sites = array_map( 'get_object_vars', get_sites( array( 'deleted' => 0 ) ) );
-
-	if ( is_array( $sites ) && $sites !== array() ) {
-		$dropdown_input = array(
-			'-' => __( 'None', 'wordpress-seo' ),
-		);
-
-		foreach ( $sites as $site ) {
-			$dropdown_input[ $site['blog_id'] ] = $site['blog_id'] . ': ' . $site['domain'];
 
-			$blog_states = array();
-			if ( $site['public'] === '1' ) {
-				$blog_states[] = __( 'public', 'wordpress-seo' );
+function wpseo_ms_maybe_restore_site() {
+	if ( isset( $_POST['wpseo_restore_blog'] ) ) {
```

Comment:
Lots of complexity / if-checks... Can't this perhaps be done in a different way (in a class function?) and returned here?

Reply:
I made the code better readable by turning around if the if checks. I agree more improvements can be made here, but since all functions in this file are temporary and will be refactored in #9688 and #9689 respectively, I'd defer until then.